carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -0,0 +1,644 @@
+      lanelet::Locations::Germany, lanelet::Participants::VehicleCar);
+  lanelet::routing::RoutingGraphUPtr shortest_path_graph =
+      lanelet::routing::RoutingGraph::build(*shortest_path_view_, *traffic_rules);
+
+  std::vector<lanelet::LineString3d> lineStrings;  // List of continuos line strings representing segments of the route
+                                                   // reference line
+
+  bool first = true;
+  size_t next_index = 0;
+
+  // Iterate over each lanelet in the shortest path this loop works by looking one lanelet ahead to detect lane changes
+  for (lanelet::ConstLanelet ll : shortest_path)
+  {
+    next_index++;
+    if (first)
+    {  // For the first lanelet store its centerline and length
+      lineStrings.push_back(copyConstructLineString(ll.centerline()));
+      first = false;
+    }
+    if (next_index < shortest_path.size())
+    {  // Check for remaining lanelets
+      auto nextLanelet = shortest_path[next_index];
+      lanelet::LineString3d nextCenterline = copyConstructLineString(nextLanelet.centerline());
+
+      size_t connectionCount = shortest_path_graph->possiblePaths(ll, 1, false).size();
+      if (connectionCount == 1)
+      {  // Get list of connected lanelets without lanechanges. On the shortest path this should only return 1 or 0
+        // No lane change
+        // Append distance to current centerline
+        lineStrings.back().insert(lineStrings.back().end(), nextCenterline.begin(), nextCenterline.end());
+      }
+      else if (connectionCount == 0)
+      {
+        // Lane change required
+        // Break the point chain when a lanechange occurs
+        lanelet::LineString3d empty_linestring;
+        empty_linestring.setId(lanelet::utils::getId());
+        distance_map.pushBack(lanelet::utils::to2D(lineStrings.back()));
+        lineStrings.push_back(empty_linestring);
+      }
+      else
+      {
+        assert(false);  // It should not be possable to reach this point. Doing so demonstrates a bug
+      }
+    }
+  }
+  // Copy values to member variables
+  shortest_path_centerlines_ = lineStrings;
+  shortest_path_distance_map_ = distance_map;
+
+  // Add length of final sections
+  if (shortest_path_centerlines_.size() > shortest_path_distance_map_.size())
+  {
+    shortest_path_distance_map_.pushBack(lanelet::utils::to2D(lineStrings.back()));  // Record length of last continuous
+                                                                                     // segment
+  }
+
+  shortest_path_filtered_centerline_view_ = lanelet::utils::createMap(shortest_path_centerlines_);
+}
+
+LaneletRoutingGraphConstPtr CARMAWorldModel::getMapRoutingGraph() const
+{
+  return std::static_pointer_cast<const lanelet::routing::RoutingGraph>(map_routing_graph_);  // Cast pointer to const
+                                                                                              // variant
+}
+
+// NOTE: See header file for details on source of logic in this function
```

Comment:
I might just have missed it but I don't see the note in the header file that's mentionedh ere.

Reply:
It's in the WorldModel.h file but I can see how this note is unclear. I'll make it more explicit.